controlplane/kubeadm/controllers/controller.go

```diff
@@ -305,6 +309,12 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	numMachines := len(ownedMachines)
 	desiredReplicas := int(*kcp.Spec.Replicas)
 
+	if numMachines > 0 && controlPlane.UnhealthyMachines().Len() == 0 {
```

Comment:
Am I wrong or by moving `reconcileHealth` from ScaleUp/ScaleDown to here we are removing it from `upgradeControlPlane`, which is called above of this point.
If yes, should we move this before L290?